ts/src/header-validator/validate-json.ts

```diff
@@ -1361,6 +1433,32 @@ function eventTriggerData(
   )
 }
 
+function attributionScopes(
+  ctx: RegistrationContext,
+  j: Json,
+  allowEmpty?: boolean
```

Comment:
I would prefer that we make this parameter required, so it is clear at each callsite what the behavior is.